### ISSUE TITLE
Why did I stop training halfway through?

### ISSUE THREAD

**Issue by LeleLucifer** [ultralytics/ultralytics#18297]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

![image](https://github.com/user-attachments/assets/3f778b8d-1263-4294-9282-f6075b7ddf89)
Why did I stop training halfway through?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @LeleLucifer, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com) for an in-depth overview of our features and workflows. Many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples are available there to guide you, and frequently asked questions may already be addressed.

If this is a 🐛 Bug Report, it will be incredibly helpful if you could provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/), including the exact command or script used, as well as the output logs leading up to the issue. This information enables us to debug effectively.

If you're asking a ❓ Question regarding custom training, please include as much detail as possible (dataset specifics, relevant code snippets, training logs, etc.) to give us adequate context. Refer to our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to ensure you're following the recommended practices.

### Upgrade

To rule out any issues resolved in the latest version, we suggest upgrading the `ultralytics` package in your environment. Make sure you're using Python>=3.8 with PyTorch>=1.8. Here's how to upgrade:

```bash
pip install -U ultralytics
```

### Environments

We also recommend testing your workflow in one of the verified environments where dependencies like Python, CUDA, and PyTorch are preconfigured for compatibility with YOLO:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image:** [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

### Community Links

For additional discussions and real-time troubleshooting:

- Connect with the community on [Discord](https://discord.com/invite/ultralytics) 🎧.
- Start threads on our [Subreddit](https://reddit.com/r/Ultralytics) for extended conversations.
- Participate in in-depth discussions at [Discourse](https://community.ultralytics.com).

### Status

Finally, check the CI status of the repository here:  
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>  
If this badge is green, it ensures all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing and validated on macOS, Windows, and Ubuntu.

Thank you for your patience 🙏! This is an automated response, but rest assured that an Ultralytics engineer will review your issue and assist you soon! 😊

**Comment by bathonSpidey** [comment]
This can happen if your batch size is high and you run out of GPU memory.

**Comment by glenn-jocher** [comment]
@bathonSpidey it is indeed likely caused by running out of GPU memory. Try reducing the batch size or using `batch=-1` to automatically determine the optimal size for your GPU. For further guidance, you can refer to the [YOLO Common Issues documentation](https://docs.ultralytics.com/guides/yolo-common-issues/).

**Comment by Skillnoob** [comment]
You should try setting `batch=-1` as mentioned above. Can you also make sure you're on the latest ultralytics version by running `pip install ultralytics -U`.
